src/repeated_game.jl

```diff
@@ -0,0 +1,368 @@
+
+    # Get number of actions for each player and create action space
+    nA1, nA2 = num_actions(p1), num_actions(p2)
+    nAS = nA1 * nA2
+    AS = QuantEcon.gridmake(1:nA1, 1:nA2)
+
+    # Create the unit circle, points, and hyperplane levels
+    C, H, Z = initialize_sg_hpl(rpd, nH)
+    Cnew = copy(C)
+
+    # Create matrices for linear programming
+    c, A, b = initialize_LP_matrices(rpd, H)
+
+    # bounds on w are [-Inf, Inf] while bounds on slack are [0, Inf]
+    lb = [-Inf, -Inf]
+    ub = [Inf, Inf]
+
+    # Set iterative parameters and iterate until converged
+    iter, dist = 0, 10.0
+    while (iter < maxiter) & (dist > tol)
+        # Compute the current worst values for each agent
+        _w1 = worst_value_1(rpd, H, C)
+        _w2 = worst_value_2(rpd, H, C)
+
+        # Update all set constraints -- Copies elements 1:nH of C into b
+        copy!(b, 1, C, 1, nH)
+
+        # Iterate over all subgradients
+        for ih=1:nH
+            #
+            # Subgradient specific instructions
+            #
+            h1, h2 = H[ih, :]
+
+            # Put the right objective into c (negative because want maximize)
+            c[1] = -h1
+            c[2] = -h2
+
+            # Allocate space to store all solutions
+            Cia = Array(Float64, nAS)
+            Wia = Array(Float64, 2, nAS)
+            for ia=1:nAS
+                #
+                # Action specific instruction
+                #
+                a1, a2 = AS[ia, :]
+
+                # Update incentive constraints
+                b[nH+1] = (1-delta)*flow_u_1(rpd, a1, a2) -
+                          (1-delta)*best_dev_payoff_1(rpd, a2) - delta*_w1
+                b[nH+2] = (1-delta)*flow_u_2(rpd, a1, a2) -
+                          (1-delta)*best_dev_payoff_2(rpd, a1) - delta*_w2
+
+                # Solve corresponding linear program
+                lpout = linprog(c, A, '<', b, lb, ub, ClpSolver())
+                if lpout.status == :Optimal
+                    # Pull out optimal value and compute
+                    w_sol = lpout.sol
+                    value = (1-delta)*flow_u(rpd, a1, a2) + delta*w_sol
+
+                    # Save hyperplane level and continuation promises
+                    Cia[ia] = h1*value[1] + h2*value[2]
+                    Wia[:, ia] = value
+                else
+                    Cia[ia] = -Inf
+                end
+            end
+
+            any(Cia .> -1e12) ? nothing : warn("All actions result in infeasiblility")
+            # Action which pushes furthest in direction h_i
+            astar = indmax(Cia)
+            a1star, a2star = AS[astar, :]
+
+            # Get hyperplane level and continuation value
+            Cstar = Cia[astar]
+            Wstar = Wia[:, astar]
+            if Cstar > -Inf
+                Cnew[ih] = Cstar
+            else
+                warn("Failed to find hyperplane level greater than -Inf")
+                Cnew[ih] = h1*min_pane_payoffs[1] + h2*min_pane_payoffs[2]
```

Comment:
Generally, the minmax action profile is not a Nash equilibrium. What happens if these values are replaced by the minimax payoffs?